command/deployment_status.go

```diff
@@ -37,6 +46,9 @@ Status Options:
   -json
     Output the deployment in its JSON format.
 
+  -monitor
+    Enter monitor mode directly without modifying the deployment status.
```

Comment:
The wording here almost implies that `nomad deployment status` could otherwise modify the deployment. And it doesn't tell the end user why they'd want to "monitor mode". What if we wrote something like the following?

```suggestion
    Enter monitor mode to poll for updates to the deployment status.
```

We'll want to add this flag to the documentation as well. That's in `./website/content/docs/commands/deployment/status.mdx`